src/jdk.jpackage/share/classes/jdk/jpackage/internal/AppImageFile.java

```diff
@@ -332,29 +298,43 @@ private static String xpathQueryNullable(XPath xPath, String xpathExpr,
     }
 
     static String getVersion() {
-        return "1.0";
+        return System.getProperty("java.version");
     }
 
     static String getPlatform() {
         return PLATFORM_LABELS.get(Platform.getPlatform());
     }
 
     private boolean isValid() {
+        if (!Objects.equals(getVersion(), creatorVersion)) {
+            return false;
+        }
+
+        if (!Objects.equals(getPlatform(), creatorPlatform)) {
+            return false;
+        }
+
         if (launcherName == null || launcherName.length() == 0) {
             return false;
         }
 
+        if (mainClass == null || mainClass.length() == 0) {
+            return false;
+        }
+
         for (var launcher : addLauncherInfos) {
             if ("".equals(launcher.getName())) {
                 return false;
             }
         }
 
-        if (!Objects.equals(getVersion(), creatorVersion)) {
+        if (signedStr == null ||
+                !("true".equals(signedStr) || "false".equals(signedStr))) {
             return false;
         }
 
-        if (!Objects.equals(getPlatform(), creatorPlatform)) {
+        if (appStoreStr == null ||
+                !("true".equals(appStoreStr) || "false".equals(appStoreStr))) {
             return false;
         }
```

Comment:
It makes sense to unfold this function in the ctor as we don't allow empty AppImageFile instances.

Reply:
Done.